src/index.js

```diff
@@ -1,150 +1,140 @@
+        element.innerHTML = getCityTimeHtmlTemplate('', '', 1)
+        topContainer.appendChild(element)
+      }
     }
   }
-}
 
-/**
- *This function will change values if the option is changed in city selection
- */
-function changeValues () {
-  const selectedCity = document.getElementsByClassName('option-select')[0].value.toLowerCase()
-  console.log(selectedCity)
-  document.getElementById('top-img').src = '../Assets/HTML&CSS/Icons_for_cities/' + selectedCity + '.svg'
-  console.log(allCityDetails[selectedCity].temperature)
-  document.getElementById('tempc-val').innerHTML = allCityDetails[selectedCity].temperature
-  document.getElementById('tempf-val').innerHTML = ((parseInt(allCityDetails[selectedCity].temperature) * 9 / 5) + 32).toFixed(2) + 'F'
-  document.getElementById('humid-val').innerHTML = allCityDetails[selectedCity].humidity
-  document.getElementById('precip-val').innerHTML = allCityDetails[selectedCity].precipitation
-  const time = document.getElementsByClassName('time-text')[0]
-  const date = document.getElementById('date-text')
-  const timeJson = allCityDetails[selectedCity].dateAndTime
-  console.log(timeJson)
-  const separatedArray = timeJson.split(', ')
-  console.log(separatedArray)
-  const months = ['JAN', 'FEB', 'MAR', 'APR', 'MAY', 'JUN', 'JUL', 'AUG', 'SEP', 'OCT', 'NOV', 'DEC']
-  const splitDate = separatedArray[0].split('/')
-  console.log(splitDate)
-  date.innerHTML = splitDate[1] + '-' + months[parseInt(splitDate[0]) - 1] + '-' + splitDate[2]
-  console.log(date)
-  const splitTime = separatedArray[1].split(':')
-  console.log(splitTime)
-  const splitSecAndMeridiem = splitTime[2].split(' ')
-  time.innerHTML = splitTime[0] + ':' + splitTime[1] + ':' + '<small>' + splitSecAndMeridiem[0] + '<small>'
-  const timeState = document.getElementById('time-img')
-  console.log(splitSecAndMeridiem[1])
-  if (splitSecAndMeridiem[1] === 'PM') {
-    timeState.src = '../Assets/HTML&CSS/GeneralImages&Icons/pmState.svg'
-    time.classList.add('pm-state')
-  } else {
-    timeState.src = '../Assets/HTML&CSS/GeneralImages&Icons/amState.svg'
-    time.classList.add('time-text')
-    time.classList.remove('pm-state')
+  /**
+   *This function will change values if the option is changed in city selection
+   */
+  function changeHeaderValues () {
+    const selectedCity = document.getElementsByClassName('option-select')[0].value.toLowerCase()
+    document.getElementById('top-img').src = '../Assets/HTML&CSS/Icons_for_cities/' + selectedCity + '.svg'
+    const cityDetails = allCityDetails[selectedCity]
+    document.getElementById('tempc-val').innerHTML = cityDetails.temperature
+    document.getElementById('tempf-val').innerHTML = ((parseInt(cityDetails.temperature) * 9 / 5) + 32).toFixed(2) + 'F'
+    document.getElementById('humid-val').innerHTML = cityDetails.humidity
+    document.getElementById('precip-val').innerHTML = cityDetails.precipitation
+    const time = document.getElementsByClassName('time-text')[0]
```

Comment:
All the variables declaration can be grouped together and placed at the top of it's scope if possible. 
i.e all variables inside the function at first and there computations after that.